### PR TITLE
`CONTRIBUTING.md`: add reference to `util/check-format.pl` and fix several nits

### PR DESCRIPTION
**Pull request by DDvO**
Add the following recommendation:

> There is a [Perl tool](util/check-format.pl) that helps finding code formatting mistakes and other coding style nits.

Also add two links as well as a few minor generalisations and corrections, and fix a couple of nits.

### CODE REVIEW

**Comment by DDvO** [comment]
CI failure on `80-test_cmp_http.t ` with sanitizers is clearly unrelated (and appears intermittent).

**Review by hlandau — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -64,22 +67,24 @@ guidelines:
...
- 4. Patches should follow our [coding style] and compile without warnings.
+ 4. Code provided should follow our [coding style] and compile without warnings.
+    There is a [Perl tool](util/check-format.pl) that helps
+    finding code formatting mistakes and other coding style nits.
     Where `gcc` or `clang` is available, you should use the
     `--strict-warnings` `Configure` option.  OpenSSL compiles on many varied
     platforms: try to ensure you only use portable features.  Clean builds via
-    GitHub Actions and AppVeyor are required, and they are started automatically
-    whenever a PR is created or updated.
+    GitHub Actions and AppVeyor are required. They are started automatically
```
Comment (levitte) [inline CONTRIBUTING.md:76]: Er...  we've dropped Appveyor in favor of Github Actions a loooong time ago.  It should have been removed from this file then.  Would you please remove it now?

**Comment by t8m** [comment]
This should go into all branches. This is a doc fix.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master, 3.2, 3.1, and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**